src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java

```diff
@@ -17,9 +18,26 @@ public void execute(Job2dDriver driver) {
             command.execute(driver);
         }
     }
-
     @Override
     public Iterator<DriverCommand> iterator() {
         return commandList.iterator();
     }
+
+    @Override
+    protected ComplexCommand clone() throws CloneNotSupportedException {
+        return (ComplexCommand) super.clone();
+    }
+
+    @Override
+    public Object copyDriverCommand() throws CloneNotSupportedException {
+        return ICompoundCommand.super.copyDriverCommand();
+    }
+
+    public ComplexCommand copy() throws CloneNotSupportedException {
+        List<DriverCommand> newCommandList=new ArrayList<>();
+        for(DriverCommand driverCommand: this.commandList){
+            newCommandList.add((DriverCommand) driverCommand.copyDriverCommand());
+        }
+        return new ComplexCommand(newCommandList);
+    }
```

Comment:
Another public method responsible for copy operation. I suggest leaving only one in the DriverCommand interface (copy() would be shorter and still ok).